prime-router/src/main/kotlin/azure/DatabaseAccess.kt

```diff
@@ -367,19 +368,38 @@ class DatabaseAccess(private val create: DSLContext) : Logging {
     }
 
     /**
-     * Fetch an action ID for a given [reportId].
+     * Fetch an action for a given [reportId].
      * @param txn an optional database transaction
-     * @return an action ID, or null if no action ID was found
+     * @return an Action ID, or null if no such reportId exists.
      */
-    fun fetchActionIdForReport(
+    fun fetchActionForReportId(
         reportId: UUID,
```

Comment:
This query takes a reportID and gets its Action obj.    The full Action obj is needed now to do proper authorization (does the user have the rights to see this action and its report?)